diesel/src/mysql/connection/url.rs

```diff
@@ -110,3 +115,18 @@ fn first_path_segment_is_treated_as_database() {
         ConnectionOptions::parse("mysql://localhost").unwrap().database()
     );
 }
+
+#[test]
+fn userinfo_should_be_percent_decode() {
+    let mut db_url = Url::parse("mysql://localhost/bar").unwrap();
+    let username = "x#gfuL?4Zuj{n73m}eeJt0";
+    let password = "x/gfuL?4Zuj{n73m}eeJt1";
+    db_url.set_username(username).unwrap();
+    db_url.set_password(Some(password)).unwrap();
```

Comment:
I'm finding this test rather difficult to read. Rather than relying on the (implicit to this test) behavior of `set_username` and `set_password` doing percent encoding, can we just have a URL string which includes percent encoded/decoded characters?

Reply:
Sure, updated